perllib/FixMyStreet/Cobrand/Bexley/Waste.pm

```diff
@@ -56,6 +56,25 @@ use constant WHITESPACE_UNDEF_DATE => '0001-01-01T00:00:00';
 use constant MISSED_COLLECTION_SERVICE_PROPERTY_ID => 68;
 use constant REQUEST_SERVICE_PROPERTY_ID => 69;
 
+sub fetch_whitespace_data {
+    my ($self, $method, $uprn) = @_;
+
+    my $c = $self->{c};
+
+    # Check if we've already fetched the data for this UPRN
+    if (!$c->stash->{whitespace_data} || $c->stash->{whitespace_data_uprn} ne $uprn) {
```

Comment:
Couple of questions/more info:
1. call_api is already caching returned results (by using waste_cache_get/set), so do you need to be storing anything in the stash at all - can you just call call_api and return the data, letting it do the caching? Or is the stash doing something else here?
2. Rather than have a `whitespace_data_uprn` check, would it be better to have `whitespace_data` be a UPRN-keyed hash? Then you could have if $c->stash->{whitespace_data}{$uprn} doesn't exist, call_api and store it there, and return $c->stash->{whitespace_data}{$uprn}{"$method $uprn"}. Guess it doesn't make much difference.